ui/src/app/workflows/components/workflow-logs-viewer/workflow-logs-viewer.tsx

```diff
@@ -222,6 +229,7 @@ export function WorkflowLogsViewer({workflow, nodeId, initialPodName, container,
                     value={(podNames.find(x => x.value === podName) || {label: ''}).label}
                     onSelect={(_, item) => {
                         setPodName(item.value);
+                        setNodeId(podNamesToNodeIDs.get(item.value)); // set the correct node id to be able to fetch archived logs
```

Comment:
this can still be a derivation as I mentioned [before](https://github.com/argoproj/argo-workflows/pull/12591?discussion_r1538305749) -- just derived based on `podName` and `nodeId`, if I'm understanding correctly

Reply:
I derived it in [this commit](https://github.com/argoproj/argo-workflows/pull/12591/commits/b58a666f49eaf96a2059f8170af0cd12b89ebffa), which is much cleaner and avoids all the state and logic above too